skbio/sequence/__init__.py

```diff
@@ -24,7 +24,7 @@
 
    Sequence
    IUPACSequence
-   NucleotideSequence
+   NucleotideMixin
```

Comment:
I'm ok with hiding it. Could we also just drop it from our docs? Then it is there, but our support only goes as far as what you are willing to read from the source code docs